From 8c69a7eb85f4a2e61295df413a4e631732629122 Sun, 18 Oct 2015 14:29:34 -0400
From: Michael Booth <m.judd.booth@gmail.com>
Date: Thu, 23 Apr 2015 23:03:10 -0400
Subject: [PATCH] use nanfuncs rather than explicit np dependence


diff --git a/pyqtgraph/graphicsItems/PlotCurveItem.py b/pyqtgraph/graphicsItems/PlotCurveItem.py
index 3d3e969..fef1c99 100644
--- a/pyqtgraph/graphicsItems/PlotCurveItem.py
+++ b/pyqtgraph/graphicsItems/PlotCurveItem.py
@@ -12,6 +12,8 @@
 import struct, sys
 from .. import getConfigOption
 from .. import debug
+from ..util import nanfuncs
+
 
 __all__ = ['PlotCurveItem']
 class PlotCurveItem(GraphicsObject):
@@ -126,11 +128,11 @@
 
         ## Get min/max (or percentiles) of the requested data range
         if frac >= 1.0:
-            b = (np.nanmin(d), np.nanmax(d))
+            b = (nanfuncs.nanmin(d), nanfuncs.nanmax(d))
         elif frac <= 0.0:
             raise Exception("Value for parameter 'frac' must be > 0. (got %s)" % str(frac))
         else:
-            mask = np.isfinite(d)
+            mask = nanfuncs.isfinite(d)
             d = d[mask]
             b = np.percentile(d, [50 * (1 - frac), 50 * (1 + frac)])
 
